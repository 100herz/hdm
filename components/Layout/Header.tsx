import Link from "next/link"
import styled from "styled-components"

const Section = styled.section`
  position: relative;

  &::before {
    position: absolute;
    top: 0;
    left: 0;
    width: calc(2.9333 * 90%);
    height: 90%;
    background: url(/ci/hdm-bg.svg) no-repeat;
    opacity: 0.05;
    content: "";
  }

  @media (min-width: 768px) {
    position: static;

    &::before {
      left: 6rem;
      width: calc(2.9333 * 100%);
      height: 100%;
    }
  }
`

type Props = {
  className?: string
}

const Header: React.FC<Props> = ({ className = "" }: Props) => {
  return (
    <header className={`bg-gray-100 relative overflow-x-hidden ${className}`}>
      <div className="container justify-between my-6 lg:flex">
        <h1 className="pt-8">
          <img className="w-56 mx-auto" src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" />
          <span className="sr-only">HDM - Holz Design Moers GmbH</span>
        </h1>

        <nav className="flex justify-between my-16 font-semibold lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end">
          <Link href="/#produkte">
            <a>Produkte</a>
          </Link>
          <Link href="/#unternehmen">
            <a>Unternehmen</a>
          </Link>
          <Link href="/#kontakt">
            <a>Kontakt</a>
          </Link>
        </nav>
      </div>

      <Section className="container pt-12 pb-20 text-xl leading-10 lg:text-3xl lg:py-40">
        <p className="lg:max-w-3xl lg:ml-24">
          <strong>Produkt. Idee. Erfolg.</strong> Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in
          Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
        </p>
      </Section>

      <nav className="justify-around hidden px-20 my-8 text-lg font-semibold lg:flex">
        <Link href="/#crossflex">
          <a className="px-2">Crossflex</a>
        </Link>
        <Link href="/#purity">
          <a className="px-2">Purity</a>
        </Link>
        <Link href="/#cerastar">
          <a className="px-2 border-b-4 border-hdm">Cerastar</a>
        </Link>
        <Link href="/#massivholztueren">
          <a className="px-2">Massivholztüren</a>
        </Link>
        <Link href="/#moebelbauplatten">
          <a className="px-2">Möbelbauplatten</a>
        </Link>
        <Link href="/#dazu-zubehoer">
          <a className="px-2">Dazu Zubehör</a>
        </Link>
      </nav>
    </header>
  )
}

export default Header
