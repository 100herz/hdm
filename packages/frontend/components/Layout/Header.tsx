import NavLink from 'next/link'
import { useRouter } from 'next/router'
import { Link } from 'react-scroll'
import Sticky from 'react-stickynode'
import styled from '@emotion/styled'

import { useWindowDimensions } from '@/hooks/use-window-dimensions'

export default function Header({ className = '' }: React.HTMLAttributes<HTMLElement>) {
  const router = useRouter()
  const { width = 0 } = useWindowDimensions()

  return (
    <header className={`relative overflow-x-hidden ${className}`}>
      <div className="justify-between mt-6 lg:container lg:flex">
        <h1 className="pt-8 mb-10 lg:mb-0">
          <NavLink href="/">
            <a>
              <img className="w-56 mx-auto" src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" />
              <span className="sr-only">HDM - Holz Design Moers GmbH</span>
            </a>
          </NavLink>
        </h1>

        <Sticky enabled={width < 1024} innerZ="101">
          {router && router.pathname === '/' ? (
            <nav
              id="landing-main-nav"
              className="flex justify-between py-3 font-semibold bg-white sm:justify-around lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0"
            >
              <Link
                to="produkte"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-120}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Produkte
              </Link>
              <Link
                to="unternehmen"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-60}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Unternehmen
              </Link>
              <Link
                to="kontakt"
                activeClass="active"
                spy={true}
                smooth={true}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Kontakt
              </Link>
            </nav>
          ) : (
            <nav className="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0">
              <NavLink href="/#produkte">
                <a className="px-4 py-2 my-2 hover:text-hdm">Produkte</a>
              </NavLink>
              <NavLink href="/#unternehmen">
                <a className="px-4 py-2 my-2 hover:text-hdm">Unternehmen</a>
              </NavLink>
              <NavLink href="/#kontakt">
                <a className="px-4 py-2 my-2 hover:text-hdm">Kontakt</a>
              </NavLink>
            </nav>
          )}
        </Sticky>
      </div>

      <Section className="container pt-12 pb-20 lg:flex lg:flex-row lg:justify-between lg:pt-32 lg:pb-24">
        <p className="text-xl leading-10 lg:text-3xl lg:max-w-3xl lg:ml-24">
          <strong>Produkt. Idee. Erfolg.</strong> Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in
          Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
        </p>
        {router && router.pathname === '/' ? (
          <Link to="purity" activeClass="active" spy={true} smooth={true} offset={-25} duration={700}>
            <figure id="landing-eco" className="hidden -mt-16 text-center group lg:block">
              <img
                className="w-56 transition-all duration-500 transform -rotate-10 group-hover:rotate-0"
                src="/certificates/hdm-oekosiegel-header.png"
                alt="HDM - Verantwortung für Mensch und Natur"
                title="Mehr über die den BIO ... logischen Laminatboden erfahren"
              />
            </figure>
          </Link>
        ) : (
          <figure className="hidden -mt-16 text-center lg:block">
            <img
              className="w-56"
              src="/certificates/hdm-oekosiegel-header.png"
              alt="HDM - Verantwortung für Mensch und Natur"
              title="Mehr über die den BIO ... logischen Laminatboden erfahren"
            />
          </figure>
        )}
      </Section>

      {router && router.pathname === '/' && (
        <Sticky enabled={width >= 1024} innerZ="101" activeClass="sticky-nav">
          <nav className="justify-around hidden h-24 px-20 py-4 text-lg font-semibold lg:flex">
            <Link
              to="crossflex"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-120}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Dekorpaneele
            </Link>
            <Link
              to="purity"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-25}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Laminatb&ouml;den
            </Link>
            <Link
              to="cerastar"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Designb&ouml;den
            </Link>
            <Link
              to="massivholztueren"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Massivholzt&uuml;ren
            </Link>
            <Link
              to="shelves"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Möbelbauteile
            </Link>
            <Link
              to="strips"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Leisten
            </Link>
            <Link
              to="accessories"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              {!process.browser ? 'DAZU-Zubehör' : width < 1280 ? 'Zubehör' : 'DAZU-Zubehör'}
            </Link>
          </nav>
        </Sticky>
      )}
    </header>
  )
}

const Section = styled.section`
  position: relative;

  &::before {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: -2;
    background-color: #f7fafc;
    content: '';
  }

  &::after {
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
    width: calc(2.9333 * 90%);
    height: 90%;
    background-image: url(/ci/hdm-bg.svg);
    background-repeat: no-repeat;
    opacity: 0.05;
    content: '';
  }

  @media (min-width: 1024px) {
    position: static;

    &::after {
      left: 6rem;
      width: calc(2.9333 * 100%);
      height: 100%;
    }
  }
`
