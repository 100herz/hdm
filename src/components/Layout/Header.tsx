import NavLink from "next/link"
import { useRouter } from "next/router"
import { Link } from "react-scroll"
import Sticky from "react-stickynode"
import styled from "styled-components"

import { useWindowDimensions } from "@/hooks/useWindowDimensions"

const Section = styled.section`
  position: relative;

  &::before {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: -2;
    background-color: #f7fafc;
    content: "";
  }

  &::after {
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
    width: calc(2.9333 * 90%);
    height: 90%;
    background-image: url(/ci/hdm-bg.svg);
    background-repeat: no-repeat;
    opacity: 0.05;
    content: "";
  }

  @media (min-width: 768px) {
    position: static;

    &::after {
      left: 6rem;
      width: calc(2.9333 * 100%);
      height: 100%;
    }
  }
`

type Props = {
  className?: string
}

const Header: React.FC<Props> = ({ className = "" }: Props) => {
  const router = useRouter()
  const { width = 0 } = useWindowDimensions()

  return (
    <header className={`relative overflow-x-hidden ${className}`}>
      <div className="justify-between mt-6 lg:container lg:flex">
        <h1 className="pt-8 mb-10 lg:mb-0">
          <NavLink href="/">
            <a>
              <img className="w-56 mx-auto" src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" />
              <span className="sr-only">HDM - Holz Design Moers GmbH</span>
            </a>
          </NavLink>
        </h1>

        <Sticky enabled={width < 768} innerZ="101">
          {router && router.pathname === "/" ? (
            <nav className="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0">
              <Link
                to="produkte"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-120}
                duration={600}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Produkte
              </Link>
              <Link
                to="unternehmen"
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-60}
                duration={1000}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Unternehmen
              </Link>
              <Link
                to="kontakt"
                activeClass="active"
                spy={true}
                smooth={true}
                duration={1200}
                className="px-4 py-2 my-2 hover:text-hdm"
              >
                Kontakt
              </Link>
            </nav>
          ) : (
            <nav className="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0">
              <NavLink href="/#produkte">
                <a className="px-4 py-2 my-2 hover:text-hdm">Produkte</a>
              </NavLink>
              <NavLink href="/#unternehmen">
                <a className="px-4 py-2 my-2 hover:text-hdm">Unternehmen</a>
              </NavLink>
              <NavLink href="/#kontakt">
                <a className="px-4 py-2 my-2 hover:text-hdm">Kontakt</a>
              </NavLink>
            </nav>
          )}
        </Sticky>
      </div>

      <Section className="container pt-12 pb-20 lg:flex lg:flex-row lg:justify-between lg:pt-32 lg:pb-24">
        <p className="text-xl leading-10 lg:text-3xl lg:max-w-3xl lg:ml-24">
          <strong>Produkt. Idee. Erfolg.</strong> Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in
          Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
        </p>
        <Link to="purity" activeClass="active" spy={true} smooth={true} offset={-25} duration={700}>
          <a className="hidden -mt-16 lg:block hover:text-green-600">
            <figure className="text-center group">
              <img
                className="w-56 transition-all duration-500 transform -rotate-10 group-hover:rotate-0"
                src="/certificates/hdm-oekosiegel-header.png"
                alt="HDM - Verantwortung für Mensch und Natur"
                title="Mehr über die den BIO ... logischen Laminatboden erfahren"
              />
            </figure>
          </a>
        </Link>
      </Section>

      {router && router.pathname === "/" && (
        <Sticky enabled={width >= 768} innerZ="101" activeClass="sticky-nav">
          <nav className="justify-around hidden h-24 px-20 py-4 text-lg font-semibold lg:flex">
            <Link
              to="crossflex"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-120}
              duration={600}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Dekorpaneele
            </Link>
            <Link
              to="purity"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-25}
              duration={700}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Laminatb&ouml;den
            </Link>
            <Link
              to="cerastar"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={900}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Designb&ouml;den
            </Link>
            <Link
              to="massivholztueren"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-40}
              duration={1000}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              Massivholzt&uuml;ren
            </Link>
            <Link
              to="accessories"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1100}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              {width >= 768 ? "Regalböden / Möbelbauplatten" : "Böden"}
            </Link>
            <Link
              to="accessories"
              activeClass="active"
              spy={true}
              smooth={true}
              offset={-20}
              duration={1100}
              className="px-4 py-2 my-2 hover:text-hdm"
            >
              DAZU<sup>&reg;</sup> Zubeh&ouml;r
            </Link>
          </nav>
        </Sticky>
      )}
    </header>
  )
}

export default Header
