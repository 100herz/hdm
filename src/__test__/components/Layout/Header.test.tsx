import Header from "@/components/Layout/Header"
import { withTestRouter } from "@/__test__/__hoc__/withTestRouter"
import { mount, shallow } from "enzyme"
import toJSON from "enzyme-to-json"

describe("Header Testings", () => {
  test("matches snapshot", () => {
    const wrapper = shallow(<Header />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  test("render one component with the id #landing-main-nav and #landing-eco", () => {
    const wrapper = mount(withTestRouter(<Header />, { pathname: "/" }))
    expect(wrapper.find("#landing-main-nav")).toHaveLength(1)
    expect(wrapper.find("#landing-eco")).toHaveLength(1)
  })

  test("don't render a component with the id #landing-main-nav and #landing-eco", () => {
    const wrapper = mount(withTestRouter(<Header />, { pathname: "/abc" }))
    expect(wrapper.find("#landing-main-nav")).toHaveLength(0)
    expect(wrapper.find("#landing-eco")).toHaveLength(0)
  })
})
