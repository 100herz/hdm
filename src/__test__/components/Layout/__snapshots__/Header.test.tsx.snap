// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Testings matches snapshot 1`] = `
<header
  className="relative overflow-x-hidden "
>
  <div
    className="justify-between mt-6 lg:container lg:flex"
  >
    <h1
      className="pt-8 mb-10 lg:mb-0"
    >
      <Link
        href="/"
      >
        <a>
          <img
            alt="Logo der HDM GmbH"
            className="w-56 mx-auto"
            src="/ci/hdm-logo.svg"
          />
          <span
            className="sr-only"
          >
            HDM - Holz Design Moers GmbH
          </span>
        </a>
      </Link>
    </h1>
    <Sticky
      activeClass="active"
      bottomBoundary={0}
      enableTransforms={true}
      enabled={false}
      innerClass=""
      innerZ="101"
      onStateChange={null}
      releasedClass="released"
      shouldFreeze={[Function]}
      top={0}
    >
      <nav
        className="flex justify-between py-3 font-semibold bg-white lg:bg-transparent lg:text-lg lg:container lg:mb-0 lg:space-x-12 lg:justify-end lg:mt-10 lg:py-0"
      >
        <Link
          href="/#produkte"
        >
          <a
            className="px-4 py-2 my-2 hover:text-hdm"
          >
            Produkte
          </a>
        </Link>
        <Link
          href="/#unternehmen"
        >
          <a
            className="px-4 py-2 my-2 hover:text-hdm"
          >
            Unternehmen
          </a>
        </Link>
        <Link
          href="/#kontakt"
        >
          <a
            className="px-4 py-2 my-2 hover:text-hdm"
          >
            Kontakt
          </a>
        </Link>
      </nav>
    </Sticky>
  </div>
  <Header__Section
    className="container pt-12 pb-20 lg:flex lg:flex-row lg:justify-between lg:pt-32 lg:pb-24"
  >
    <p
      className="text-xl leading-10 lg:text-3xl lg:max-w-3xl lg:ml-24"
    >
      <strong>
        Produkt. Idee. Erfolg.
      </strong>
       Die Holz Design Moers GmbH bietet ihren Handelspartnern auch in Zukunft abverkaufsstarke Produkte, Konzepte und Ideen für eine erfolgreiche Zusammenarbeit.
    </p>
    <figure
      className="hidden -mt-16 text-center lg:block"
    >
      <img
        alt="HDM - Verantwortung für Mensch und Natur"
        className="w-56"
        src="/certificates/hdm-oekosiegel-header.png"
        title="Mehr über die den BIO ... logischen Laminatboden erfahren"
      />
    </figure>
  </Header__Section>
</header>
`;
